src/nccl_ofi_net.c

```diff
@@ -1275,17 +1286,23 @@ static ncclResult_t ofi_init(ncclDebugLogger_t logFunction)
 	 * better peak BW.
 	*/
 	if (IS_EFA_PROVIDER(ofi_info_list->fabric_attr->prov_name) && !support_gdr) {
-		if (cudaGetDeviceCount(&ofi_ndevices) != cudaSuccess) {
-			NCCL_OFI_WARN("Error getting CUDA device count");
+		ofi_read_num_gpus(&ofi_ndevices);
+		if (ofi_ndevices < 0) {
 			ret = ncclUnhandledCudaError;
 			goto exit;
 		}
+		NCCL_OFI_INFO(NCCL_INIT, "Found %d GPUs on the system", ofi_ndevices);
 		if (ofi_ndevices > 1)
 			ofi_ndevices /= 2;
 		if (ofi_ndevices < 1) {
 			ret = ncclSystemError;
 			goto exit;
 		}
+		char *fi_efa_dup_devices = getenv("FI_EFA_DUP_DEVICES");
```

Comment:
Please add a comment to explain why this override is required.

Reply:
This is to handle the potential scenario where the default logic of `NUM_GPUS/2` is sufficient to full utilize the EFA bandwidth.